### DISCUSSION TITLE
Fail to provision sample to Azure

### DISCUSSION THREAD

**Discussion by dylanw-oss** [microsoft/teams-sdk#1416] [Q&A]
### Language

Javascript/Typescript

### Version

latest

### Description

After testing the application on local, trying to provision it to Azure by following this:
https://github.com/microsoft/teams-ai/tree/main/js/samples#deploy-the-bot-to-azure

It failed with below error message:

`
[arm.MissingEnvironmentVariablesError]: The program cannot proceed as the following environment variables are missing: 'SECRET_OPENAI_KEY', which are required for file: /home/.../infra/azure.parameters.json. Make sure the required variables are set either by editing the .env file '/home/.../env/.env.staging' with the correct names and values , or by setting the system environment variables with the correct names and values. If you are developing with a new project created with Teams Toolkit, running provision or debug will register correct values for these environment variables.
`

**Questions**:
1. is this 'SECRET_OPENAI_KEY' for OpenAI key
2. if I use "Azure OpenAI", should I just delete SECRET_OPENAI_KEY from azure.parameters.json, and add AZURE_OPENAI_KEY & AZURE_OPENAI_ENDPOINT and run provision? (I tried it and succeed but I'm not sure if it's working)

### Reproduction Steps

```bash
see description
```

**Comment by corinagum** [comment]
1. What sample are you running? 
2. Yes, if you are not using Open AI you can delete those instances asking for those parameters. However, SECRET should still prefix the variables for Azure Open AI variables in the Teams Toolkit files (Azure bicep, `env` folder variables, yml files). 
3. SECRET should NOT prefix the variables in your `.env` file

**Comment by dylanw-oss** [comment]
@corinagum 
1. The same LightBot sample
2. Should I make changes like this?
   1) Add AZURE_OPENAI_ENDPOINT=<> to /env/.env.staging
   2) Add SECRET_AZURE_OPENAI_KEY=<> to /env/.env.staging.user

**Comment by corinagum** [comment]
Hi @dylanw-oss 
All you should need to do is add AZURE_OPENAI_ENDPOINT and AZURE_OPENAI_KEY to your `.env` file. When you run with TTK, `SECRET_` will be appended to keys and such within the TTK flow, so you should not need to make changes to `/env/` folder. try that out and let me know how it goes.
